content/post/38-js-ipfs-0.29.md

```diff
@@ -0,0 +1,88 @@
+---
+date: 2018-05-29
+url: 38-js-ipfs-0-29
+title: js-ipfs 0.29.0 released
+author: David Dias
+---
+
+<blockquote class="twitter-tweet" data-conversation="none" data-lang="en"><p lang="en" dir="ltr">I love the smell of a new release in the morning 😁<a href="https://twitter.com/hashtag/IPFS?src=hash&amp;ref_src=twsrc%5Etfw">#IPFS</a> <a href="https://twitter.com/hashtag/JS?src=hash&amp;ref_src=twsrc%5Etfw">#JS</a> v0.29.0 🚀<br><br>✔ Test <br>✔ Build dist version <br>✔ Update Contributors list <br>✔ Bump Version: v0.28.2 -&gt; v0.29.0 <br>✔ Gen Changelog <br>✔ Publish to npm 🌟<br><br>Find the latest on your favorite module store!</p>&mdash; David Dias (@daviddias) <a href="https://twitter.com/daviddias/status/1001432008302694400?ref_src=twsrc%5Etfw">May 29, 2018</a></blockquote>
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+# 🔦 Highlights
+
+### 🏎 Bitswap is 30% faster
+
+Bitswap is now 30% thanks to fixing a bug on [js-ipfs-bitswap](https://github.com/ipfs/js-ipfs-bitswap/pull/175#issuecomment-390957244) that was causing every message to be sent individually.
+
+### 👓 js-ipfs now works with uglify-es just fine
+
+The duck typing throughout js-ipfs has been improved and that means that now you can use uglify-es again to minify your js-ipfs code. The result is that the js-ipfs bundled version is now back to its minified size as well. 
+
+### ☎️ Circuit Relay was merged in
+
+Circuit Relay enables connectivity between any two nodes through a third relay node. This ensures connectivity in almost every scenario, that is, as long as peers can connect to some public IP node. Learn how to use it [here](https://github.com/ipfs/js-ipfs/tree/master/examples/circuit-relaying). This was one of the two features that js-ipfs was missing to get full connectivity.
```

Comment:
I feel like this needs more bravado. This is BIG news! Maybe first item? More emoji? Mention NAT traversal?

Reply:
It is indeed a big deal. We should be careful to mention that we still need a go release to work out all the issues with go interop. 

RE: NAT traversal, we might want to say that our next steps are building on top of circuit with true NAT traversal with the use of UPnP and NAT-PMP, which will allow opening external ports on the users router as well as the possibility of using techniques such as `dial me back` once a ports are mapped and a circuited connection is established - all this might require a new blog post tho 😉.